src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs

```diff
@@ -63,83 +63,54 @@ internal static unsafe partial class Utf16Utility
             // ARM64-intrinsicified code paths. ARM64 platforms may still use the vectorized
             // non-intrinsicified 'else' block below.
 
-            if (/* (AdvSimd.Arm64.IsSupported && BitConverter.IsLittleEndian) || */ Sse41.IsSupported)
+            if (/* (AdvSimd.Arm64.IsSupported && BitConverter.IsLittleEndian) || */ Sse2.IsSupported)
             {
                 if (inputLength >= Vector128<ushort>.Count)
                 {
                     Vector128<ushort> vector0080 = Vector128.Create((ushort)0x0080);
                     Vector128<ushort> vector7800 = Vector128.Create((ushort)0x7800);
                     Vector128<ushort> vectorA000 = Vector128.Create((ushort)0xA000);
 
-                    Vector128<byte> bitMask128 = BitConverter.IsLittleEndian ?
-                        Vector128.Create(0x80402010_08040201).AsByte() :
-                        Vector128.Create(0x01020408_10204080).AsByte();
-
                     char* pHighestAddressWhereCanReadOneVector = pEndOfInputBuffer - Vector128<ushort>.Count;
                     Debug.Assert(pHighestAddressWhereCanReadOneVector >= pInputBuffer);
 
                     do
                     {
-                        Vector128<ushort> utf16Data;
-                        if (AdvSimd.Arm64.IsSupported)
-                        {
-                            utf16Data = AdvSimd.LoadVector128((ushort*)pInputBuffer); // unaligned
-                        }
-                        else
-                        {
-                            utf16Data = Sse2.LoadVector128((ushort*)pInputBuffer); // unaligned
-                        }
+                        Vector128<ushort> utf16Data = Vector128.Load((ushort*)pInputBuffer);
 
                         pInputBuffer += Vector128<ushort>.Count; // eagerly bump this now in preparation for next loop, will adjust later if necessary
-                        Vector128<ushort> charIsNonAscii;
 
                         // Sets the 0x0080 bit of each element in 'charIsNonAscii' if the corresponding
                         // input was 0x0080 <= [value]. (i.e., [value] is non-ASCII.)
 
-                        if (AdvSimd.Arm64.IsSupported)
+                        Vector128<ushort> charIsNonAscii;
+                        if (Sse41.IsSupported)
                         {
-                            charIsNonAscii = AdvSimd.Min(utf16Data, vector0080);
+                            // Use Sse41.Min with opportunistic ISA check on R2R instead of
+                            // slower SSE2-baked Vector128.Min
```

Comment:
Why is Vector128.Min slower than Sse41.Min?

Reply:
`Vector128.Min` uses fallback without SSE4.1:
```cs
Vector128<ushort> Min(Vector128<ushort> a, Vector128<ushort> b)
    => Vector128.Min(a, b);
```
```asm
; Method mytest:Min
       movups   xmm0, xmmword ptr [rdx]
       movups   xmm1, xmmword ptr [reloc @RWD00]
       paddw    xmm0, xmm1
       movups   xmm2, xmmword ptr [r8]
       paddw    xmm2, xmm1
       pminsw   xmm0, xmm2
       psubw    xmm0, xmm1
       movups   xmmword ptr [rcx], xmm0
       mov      rax, rcx
       ret      
RWD00  	dq	8000800080008000h, 8000800080008000h
; Total bytes of code: 37
```
it probably can be improved but Sse4.1 Min is a single instruction under a quick opportunistic check branch.

Reply:
I am not sure R2R codegen that important though

Reply:
Is that a bug in Vector128.Min then?

Reply:
This pattern is only for R2R/NAOT which is compiled as SSE2, for JIT it's not needed.
the idea to emit:
```
if (runtimeIsaCheck == 1)
{
    // use single-instruction for Min
}
else
{
    // slow fallback because there is no SSE4.1
}
```
but I decided to drop it because it looks like we have this problem already in many places where we only check `Vector128.IsHardwareAccelerated` and don't care what exactly Vector128.* APIs need under the hood.